ceh-2000/gpu_guru
# Cleaning up printing and logging

# Closes #14

This PR cleans up printing of hyperparameters by adding logic that controls printing based on the parameters that are relevant for each algorithm. It also cleans the naming for `tensorboard` run logging.

**TODO:**
- [x] Wait for the `FedProx` branch to be merged in, then ensure that logging still works well for `FedProx` and other algorithms

# Testing
1. Pull down the most recent version of the branch.
2. Delete your `/runs/` directory to clean out `tensorboard` runs.
3. Run the following and check that the printed hyperparameters are appropriate for the algorithm AND that the `tensorboard` run name includes all relevant information. 
  - `FedAvg`: run `python3 main.py --algorithm fedavg --dataset mnist --num_users 10 --alpha 0.1 --sample_ratio 0.1 --glob_epochs 5 --local_epochs 1 --should_log 1`
  - `FedProx`: run `python3 main.py --algorithm fedprox --dataset mnist --num_users 10 --alpha 0.1 --sample_ratio 0.1 --glob_epochs 5 --local_epochs 1 --should_log 1 --mu 1`
  - `Central`: run `python3 main.py --algorithm central --dataset mnist --sample_ratio 0.1 --glob_epochs 5 --should_log 1`
  - `One-shot`: run `python3 main.py --algorithm oneshot --num_users 5 --alpha 1.0 --sample_ratio 0.1 --local_epochs 5 --should_log 1 --one_shot_sampling validation --user_data_split 0.9 --K 3`
